logger/request_context_test.go

```diff
@@ -0,0 +1,55 @@
+package logger
+
+import (
+	"context"
+	"net/http"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("logger", func() {
+
+	Describe("populateRequestContext", func() {
+		It("returns a context including a RequestContext based on the given request", func() {
+			const (
+				remoteAddr = "127.0.0.1"
+				reqID      = "123"
+			)
+			r, err := http.NewRequest("GET", "https://homes.co.nz", http.NoBody)
+			Expect(err).ToNot(HaveOccurred())
+			r.RemoteAddr = remoteAddr
+			r.Header.Set("X-Request-Id", reqID)
+			ctx := populateRequestContext(context.Background(), r)
+
+			rctx := retrieveRequestContext(ctx)
+			Expect(rctx.RemoteAddr).To(Equal(remoteAddr))
+			Expect(rctx.RequestID).To(Equal(reqID))
+		})
+		Context("using X-Forwarded-For instead of remote addr", func() {
+			It("returns a context including a RequestContext based on the given request", func() {
+				const (
+					remoteAddr = "127.0.0.1"
+					reqID      = "123"
+				)
+				r, err := http.NewRequest("GET", "https://homes.co.nz", http.NoBody)
+				Expect(err).ToNot(HaveOccurred())
+				r.Header.Set("X-Forwarded-For", remoteAddr)
+				r.Header.Set("X-Request-Id", reqID)
+				ctx := populateRequestContext(context.Background(), r)
+
+				rctx := retrieveRequestContext(ctx)
+				Expect(rctx.RemoteAddr).To(Equal(remoteAddr))
+				Expect(rctx.RequestID).To(Equal(reqID))
+			})
+		})
+	})
+
+	Describe("retrieveRequestContext", func() {
+		It("returns an empty (but valid) context if not initialized", func() {
+			rctx := retrieveRequestContext(context.Background())
```

Comment:
What does logrus do in this case? I'm wondering if retrieveRequestContext should return if the fetch was okay or not? Something like this can handle both nil and an unexpected type
```go
func retrieveRequestContext(ctx context.Context) (RequestContext, bool) {
	v, ok := ctx.Value(ContextKeyRequestContext).(RequestContext)
	return v, ok
}
```

Reply:
It would log with those fields in its context, but they'd be empty. Probably better to just remove them altogether, so have updated it so FieldsFromContext explicitly returns an empty logrus.Fields.